Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/CollisionSensor.cpp

```diff
@@ -74,17 +110,94 @@ void ACollisionSensor::OnCollisionEvent(
     }
   }
 
-  const auto &Episode = GetEpisode();
+  const auto& CurrentEpisode = GetEpisode();
+  constexpr float TO_METERS = 1e-2;
```

Comment:
**Prevent code logic duplication**
I suggest to create a common function for all collision events in CollisionSensor as the logic is the same. The structure should look like this:

```c++

//This one do not need to be an UFunction I think
void ACollisionSensor::OnCollisionEvent(
    AActor *Actor,
    AActor *OtherActor,
    FVector NormalImpulse,
    const FHitResult &Hit)
{
  //Paste here the shared code
}

void ACollisionSensor::OnActorCollisionEvent(
    AActor *Actor,
    AActor *OtherActor,
    FVector NormalImpulse,
    const FHitResult &Hit)
{
  OnCollisionEvent(Actor, OtherActor, ormalImpulse, Hit);
}

void ACollisionSensor::OnComponentCollisionEvent(
      UPrimitiveComponent* HitComp,
      AActor* OtherActor,
      UPrimitiveComponent* OtherComp,
      FVector NormalImpulse,
      const FHitResult& Hit)
{
  AActor* Actor = HitComp->GetOwner();
  OnCollisionEvent(Actor, OtherActor, ormalImpulse, Hit);
 }
```

Reply:
Applied